### PR TITLE
chore: support scene number messages from renderer

### PR DESCRIPTION
**Pull request by pravusjif**
- [Added missing `sceneNumber` property in `CreateGlobalScene()` message data structure](https://github.com/decentraland/kernel/pull/588/commits/5ecd17e32a8b34cbfcda25435875fba95d5bebe7) (without it Portable Experience scenes pipeline throws a nullref exception in the renderer)
- Added `sceneNumber` support in every needed message in `browserInterface` (keeping also `sceneId` for compatibility)

Renderer-side: https://github.com/decentraland/unity-renderer/pull/2977

### CODE REVIEW

File: packages/shared/world/sagas.ts [hunk 1]
```diff
@@ -126,7 +131,10 @@ export function* anounceOnEnterOnSceneStart() {
 export function* anounceOnReadyOnSceneReady() {
   while (true) {
     const event: RendererSignalSceneReady = yield take(RENDERER_SIGNAL_SCENE_READY)
-    const scene: SceneWorker | undefined = yield call(getSceneWorkerBySceneID, event.payload.sceneId)
+    const scene: SceneWorker | undefined = yield event.payload.sceneId
+      ? call(getSceneWorkerBySceneID, event.payload.sceneId)
+      : call(getSceneWorkerBySceneNumber, event.payload.sceneNumber)
```
[the remark this answers is not in the document]
Comment (pravusjif) [reply packages/shared/world/sagas.ts]: done!

File: packages/unity-interface/BrowserInterface.ts [hunk 2]
```diff
@@ -281,8 +286,9 @@ export class BrowserInterface {
     futures[data.id].resolve(data.mousePosition)
   }
 
-  public SceneEvent(data: { sceneId: string; eventType: string; payload: any }) {
-    const scene = getSceneWorkerBySceneID(data.sceneId)
+  public SceneEvent(data: { sceneId: string; sceneNumber: number; eventType: string; payload: any }) {
+    const scene = data.sceneId ? getSceneWorkerBySceneID(data.sceneId) : getSceneWorkerBySceneNumber(data.sceneNumber)
```
[the remark this answers is not in the document]
Comment (pravusjif) [reply packages/unity-interface/BrowserInterface.ts]: done!

File: packages/unity-interface/BrowserInterface.ts [hunk 3]
```diff
@@ -1017,12 +1031,15 @@ export class BrowserInterface {
...
+    sceneNumber: number
     videoTextureId: string
     status: number
     currentOffset: number
     videoLength: number
   }) {
-    const scene = getSceneWorkerBySceneID(videoEvent.sceneId)
+    const scene = videoEvent.sceneId
+      ? getSceneWorkerBySceneID(videoEvent.sceneId)
+      : getSceneWorkerBySceneNumber(videoEvent.sceneNumber)
```
[the remark this answers is not in the document]
Comment (pravusjif) [reply packages/unity-interface/BrowserInterface.ts]: done!

File: packages/shared/analytics/types.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ export type TrackEvents = PositionTrackEvents & {
   ['referral_save']: { code: string; address?: string; referral_of?: unknown }
   ['Move to Parcel']: { newParcel: string; oldParcel: string | null; exactPosition: ReadOnlyVector3 }
   ['motd_failed']: Record<string, unknown> // {}
-  ['TermsOfServiceResponse']: { sceneId: string; accepted: boolean; dontShowAgain: boolean }
+  ['TermsOfServiceResponse']: { sceneId: string; sceneNumber: number; accepted: boolean; dontShowAgain: boolean }
```
[the remark this answers is not in the document]
Comment (pravusjif) [reply packages/shared/analytics/types.ts]: done!

File: packages/shared/world/sagas.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (pravusjif) [reply packages/shared/world/sagas.ts]: In the end I had to re-add the `yield` as it's the only way I found to keep the [pertinent test](https://github.com/decentraland/kernel/blob/3d929068e66f60af6d29ba2b14fe79c207181705/test/unit/world.saga.test.tsx#L37) passing.
I also added a new test specifically for scene number [here](https://github.com/decentraland/kernel/blob/3d929068e66f60af6d29ba2b14fe79c207181705/test/unit/world.saga.test.tsx#L18).

File: packages/unity-interface/BrowserInterface.ts [hunk 5]
```diff
@@ -564,8 +587,8 @@ export class BrowserInterface {
     getUnseenMessagesByUser()
   }
 
-  public SetHomeScene(data: { sceneId: string }) {
-    store.dispatch(setHomeScene(data.sceneId))
+  public SetHomeScene(data: { sceneCoords: string }) {
+    store.dispatch(setHomeScene(data.sceneCoords))
```
Comment (menduz) [inline packages/unity-interface/BrowserInterface.ts:591]: Will this continue to work until the unity-renderer counterpart is merged?

**Review by menduz — COMMENTED** [review]
Only one comment and ready to merge 🙌

File: packages/unity-interface/BrowserInterface.ts (hunk 5 above)
Comment (pravusjif) [reply packages/unity-interface/BrowserInterface.ts]: Nice catch, actually in production that is mistakenly called `sceneId` even though the renderer is [sending just the scene coords as a string](https://github.com/decentraland/unity-renderer/blob/dev/unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HomePoint/WebInterfaceHomeLocationController.cs#L8). And in the [counterpart renderer PR](https://github.com/decentraland/unity-renderer/pull/2977) I corrected that.

I just updated this Kernel PR to allow the data with both names, and after merging the renderer  counterpart PR we can then remove the misleading `sceneId` one here.

I tested the "set home" functionality working with both prod renderer and the counterpart PR renderer.

**Review by menduz — APPROVED** [review]
🚢 it

**Pull request merged**